src/librustc_codegen_llvm/common.rs

```diff
@@ -388,3 +386,9 @@ pub fn struct_in_context(
 fn hi_lo_to_u128(lo: u64, hi: u64) -> u128 {
     ((hi as u128) << 64) | (lo as u128)
 }
+
+fn is_const_integral(v: &'ll Value) -> bool {
+    unsafe {
+        llvm::LLVMIsAConstantInt(v).is_some()
```

Comment:
Another place where stricter types could match LLVM's hierarchy (again, not necessarily relevant to this PR): `LLVMConstIntGetZExtValue` could take a `&ConstantInt` which `LLVMIsAConstantInt` would return an `Option` thereof, making `const_to_opt_uint(v)` be `LLVMIsAConstantInt(v).map(LLVMConstIntGetZExtValue)`.